salt/states/pip_state.py

```diff
@@ -163,8 +164,11 @@ def _check_pkg_version_format(pkg):
         ret['version_spec'] = []
     else:
         ret['result'] = True
-        ret['prefix'] = install_req.req.project_name
-        ret['version_spec'] = install_req.req.specs
+        if hasattr(install_req.req, 'project_name'):
+            ret['prefix'] = install_req.req.project_name
+        else:
+            ret['prefix'] = re.sub('[^A-Za-z0-9.]+', '-', install_req.req.name)
+        ret['version_spec'] = [(spec.operator, spec.version) for spec in install_req.req.specifier]
```

Comment:
Again, still not a public API, but it *might* be more future proof (or it might not! yay private APIs) to do:

```python
if hasattr(install_req, "specifier"):
    specifier = install_req.specifier
else:
    specifier = install_req.req.specifier
ret['version_spec'] = [(spec.operator, spec.version) for spec in specifier]
```

The ``install_req.specifier`` attribute was added in pip 6.0, if you don't care about supporting older versions than that you can just omit the ``if ... else`` completely and use ``install_req.specifier`` unambiguously.